src/components/SideNavbar.tsx

```diff
@@ -40,7 +40,8 @@ export interface NavItems {
 interface navItemProp {
   name: string;
   route: string;
-  currentRoute: string;
+  activeRoute: string;
+  setActiveRoute: any;
```

Comment:
This `setActiveRoute` function is also unnecessary since the Link already does it for us.

Reply:
This is necessary. Each item in the navbar is a `<Link/>` rather than a `<NavLink/>`. The Navlink is a link but one that applies an 'active' style to the selected link. The built in styles were breaking things though and I had trouble removing them. I'm tracking the selected link manually with a state variable, and the setActiveRoute is the parent hook to allow the children to modify the parent's state. 

If you know a way to make a state variable listen to the current route that could refactor this to be cleaner

Reply:
The Link component is what will change the route state. Then just use `useLocation` to access the current route and derive the active or inactive style from that.

Reply:
Yeah, that callback wasn't to change route but just to manage the active style. I didn't realize that `useLocation` was a hook. That makes sense. I'll refactor this in the branch that works on the navbar component